### ISSUE TITLE
run_pytest_check: support & surface snapshot baseline regeneration (--snapshot-update)

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#210]
## Summary

`run_pytest_check` is used as the canonical way to run pytest in projects that adopt mcp-tools-py (e.g. `mcp-coder`). When a project uses snapshot-style tests (e.g. `pytest-textual-snapshot`, `syrupy`), **regenerating baselines** is a normal part of the workflow — but it currently has poor support/visibility through this tool, forcing a drop down to raw `python -m pytest --snapshot-update` via a shell.

## What works today

`--snapshot-update` *does* pass through `extra_args` and runs:

```python
run_pytest_check(extra_args=["-n", "0", "--snapshot-update", "tests/icoder/test_snapshots.py"])
# -> "Pytest check completed. Collected 12 tests in 7.41 seconds | ✅ Passed: 12"
```

So functionally it's possible. The problem is **the result is indistinguishable from a normal pass**.

## The gap

1. **Regeneration outcome is hidden.** A raw run reports how many baselines were written:
   ```
   3 snapshots passed. 9 snapshots updated.
   ```
   The `run_pytest_check` summary collapses this to `✅ Passed: 12`. There is no way to tell from the tool output whether baselines were rewritten, how many, or which ones — which is exactly the information you need when regenerating. This makes silent/accidental baseline churn easy to miss.

2. **Not discoverable / pass-fail framing.** The tool's docstring is oriented entirely around pass/fail checking and never mentions snapshot workflows, so the natural conclusion (and what happened in practice) is "the MCP tool isn't meant to rewrite baselines, use Bash instead." A short note + explicit support would keep the whole workflow on the MCP tool.

## Proposed change (any subset)

- Detect snapshot-plugin summary lines (`N snapshots updated`, `N snapshots passed`; syrupy's `N snapshots generated/updated`) and surface them in the returned summary, e.g. `✅ Passed: 12 | 🔄 9 baselines updated`.
- Optionally a first-class `update_snapshots: bool = False` arg that injects the right flag for the detected plugin, so callers don't have to know plugin-specific flag names.
- Document the snapshot-regeneration use case in the docstring.

## Why it matters

Baseline regeneration is a routine, deliberate action. Surfacing "N updated" turns a silent rewrite into an auditable result, and keeps snapshot-heavy projects from having to bypass the MCP tooling for part of their test workflow.

## Context

Surfaced while working on `mcp-coder` (iCoder TUI SVG snapshots). All snapshot *checking* went through `run_pytest_check` cleanly; only baseline *regeneration* dropped to raw pytest, purely because the update outcome wasn't visible through the tool.
